src/main/java/org/zone/commands/structure/region/flags/messages/greetings/ZoneFlagGreetingsSetMessageCommand.java

```diff
@@ -4,43 +4,60 @@
 import org.jetbrains.annotations.NotNull;
 import org.spongepowered.api.command.CommandResult;
 import org.spongepowered.configurate.ConfigurateException;
+import org.zone.ZonePlugin;
 import org.zone.commands.system.ArgumentCommand;
 import org.zone.commands.system.CommandArgument;
+import org.zone.commands.system.arguments.operation.AnyMatchArgument;
 import org.zone.commands.system.arguments.operation.ExactArgument;
+import org.zone.commands.system.arguments.operation.OptionalArgument;
 import org.zone.commands.system.arguments.sponge.ComponentRemainingArgument;
 import org.zone.commands.system.arguments.zone.ZoneArgument;
 import org.zone.commands.system.context.CommandContext;
 import org.zone.permissions.ZonePermission;
 import org.zone.permissions.ZonePermissions;
 import org.zone.region.Zone;
 import org.zone.region.flag.FlagTypes;
-import org.zone.region.flag.entity.player.move.greetings.GreetingsFlag;
+import org.zone.region.flag.entity.player.move.message.display.MessageDisplayType;
+import org.zone.region.flag.entity.player.move.message.display.MessageDisplayTypes;
+import org.zone.region.flag.entity.player.move.message.greetings.GreetingsFlag;
 import org.zone.utils.Messages;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
 
 public class ZoneFlagGreetingsSetMessageCommand implements ArgumentCommand {
-    public static final ExactArgument REGION = new ExactArgument("region");
-    public static final ExactArgument FLAGS = new ExactArgument("flag");
+
     public static final ZoneArgument ZONE_VALUE = new ZoneArgument("zoneId",
             new ZoneArgument.ZoneArgumentPropertiesBuilder().setBypassSuggestionPermission(
                     ZonePermissions.OVERRIDE_FLAG_GREETINGS_SET));
-    public static final ExactArgument GREETINGS = new ExactArgument("greetings");
-    public static final ExactArgument MESSAGE = new ExactArgument("message");
-    public static final ExactArgument SET = new ExactArgument("set");
     public static final ComponentRemainingArgument MESSAGE_VALUE = new ComponentRemainingArgument(
             "message_value");
+    public static final OptionalArgument<MessageDisplayType<?>> DISPLAY_MODE =
+            new OptionalArgument<>(new AnyMatchArgument<>("displayMode",
+                    MessageDisplayType::getId
+                    , ZonePlugin
+                    .getZonesPlugin()
+                    .getMessageDisplayManager()
+                    .getDisplayTypes()),
+                    MessageDisplayTypes.CHAT_MESSAGE_DISPLAY);
```

Comment:
This may confuse the user as in its current form, if there is no message, then the display mode will be set, however if there is a message then the display mode will not be set

Reply:
As mentioned before, either remove the argument or ensure that the flag gets set the display mode type